File: backend/test-multi-modelo.js
/**
 * SCRIPT DE PRUEBA - SISTEMA MULTI-MODELO
 * Prueba todos los modelos de IA disponibles
 */

import dotenv from 'dotenv';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

// Configurar dotenv
const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);
dotenv.config({ path: join(__dirname, '.env') });

import { 
  ejecutarChatInteligente,
  generarRespuestaIA, 
  obtenerEstadisticas,
  MODELOS,
  ESTRATEGIA_FALLBACK
} from './config/modelos-ia.js';

console.log('🤖 SISTEMA MULTI-MODELO DE IA - TESCHA\n');
console.log('═'.repeat(60));

// Mostrar modelos disponibles
console.log('\n📦 MODELOS DISPONIBLES:\n');

Object.keys(MODELOS).forEach(provider => {
  console.log(`\n🔹 ${provider.toUpperCase()}:`);
  Object.keys(MODELOS[provider]).forEach(modelo => {
    const info = MODELOS[provider][modelo];
    console.log(`   • ${info.nombre} (${info.id})`);
    if (info.descripcion) console.log(`     ${info.descripcion}`);
  });
});

console.log('\n═'.repeat(60));
console.log('\n🔄 ESTRATEGIA DE FALLBACK:\n');
ESTRATEGIA_FALLBACK.forEach((modelo, i) => {
  console.log(`${i + 1}. ${modelo}`);
});

console.log('\n═'.repeat(60));
console.log('\n🧪 PRUEBA 1: Respuesta Simple\n');

try {
  const inicio = Date.now();
  const respuesta = await generarRespuestaIA(
    '¿Qué es el sistema escolar TESCHA? Responde en máximo 50 palabras.',
    'Eres un asistente experto del sistema TESCHA.',
    200
  );
  const tiempo = Date.now() - inicio;
  
  console.log(`✅ Respuesta recibida en ${tiempo}ms:`);
  console.log(`\n${respuesta}\n`);
} catch (error) {
  console.error(`❌ Error: ${error.message}`);
}

console.log('═'.repeat(60));
console.log('\n🧪 PRUEBA 2: Chat con Historial\n');

try {
  const messages = [
    { 
      role: 'system', 
      content: 'Eres un asistente matemático. Responde de forma concisa.' 
    },
    { 
      role: 'user', 
      content: '¿Cuánto es 15 × 23?' 
    }
  ];
  
  const inicio = Date.now();
  const respuesta = await ejecutarChatInteligente(messages, [], 150);
  const tiempo = Date.now() - inicio;
  
  console.log(`✅ Respuesta recibida en ${tiempo}ms:`);
  console.log(`\n${respuesta.content}\n`);
} catch (error) {
  console.error(`❌ Error: ${error.message}`);
}

console.log('═'.repeat(60));
console.log('\n📊 ESTADÍSTICAS DEL SISTEMA:\n');

const stats = obtenerEstadisticas();

console.log('🔧 Configuración:');
console.log(`   • Groq API Keys: ${stats.configuracion.groqKeys}`);
console.log(`   • OpenRouter: ${stats.configuracion.openrouterConfigured ? '✅ Configurado' : '❌ No configurado'}`);
console.log(`   • Modelos disponibles: ${stats.configuracion.modelosDisponibles}`);

if (Object.keys(stats.modelos).length > 0) {
  console.log('\n📈 Uso de modelos:');
  Object.entries(stats.modelos).forEach(([modelo, data]) => {
    console.log(`\n   ${modelo}:`);
    console.log(`     • Usos: ${data.usos}`);
    console.log(`     • Éxitos: ${data.exitos}`);
    console.log(`     • Fallos: ${data.fallos}`);
    console.log(`     • Tiempo promedio: ${Math.round(data.tiempoPromedio)}ms`);
    console.log(`     • Tokens usados: ${data.tokensUsados}`);
  });
} else {
  console.log('\n⚠️ No hay estadísticas aún (primer uso)');
}

console.log('\n═'.repeat(60));
console.log('\n✨ Prueba completada! El sistema está funcionando correctamente.\n');

File: backend/ai_engine/test_multi_modelo.py
"""
SCRIPT DE PRUEBA - SISTEMA MULTI-MODELO (PYTHON)
Prueba el sistema de IA desde Python
"""

import sys
import asyncio
from modelos_ia import (
    ejecutar_chat_inteligente,
    generar_respuesta_ia,
    obtener_estadisticas,
    MODELOS,
    ESTRATEGIA_FALLBACK
)

print("🤖 SISTEMA MULTI-MODELO DE IA - TESCHA (Python)\n")
print("=" * 60)

# Mostrar modelos disponibles
print("\n📦 MODELOS DISPONIBLES:\n")

for provider, modelos in MODELOS.items():
    print(f"\n🔹 {provider.upper()}:")
    for key, info in modelos.items():
        print(f"   • {info['nombre']} ({info['id']})")
        if 'descripcion' in info:
            print(f"     {info['descripcion']}")

print("\n" + "=" * 60)
print("\n🔄 ESTRATEGIA DE FALLBACK:\n")
for i, modelo in enumerate(ESTRATEGIA_FALLBACK, 1):
    print(f"{i}. {modelo}")

print("\n" + "=" * 60)
print("\n🧪 PRUEBA 1: Respuesta Simple\n")

try:
    respuesta = generar_respuesta_ia(
        prompt="¿Qué es el sistema escolar TESCHA? Responde en máximo 50 palabras.",
        system_prompt="Eres un asistente experto del sistema TESCHA.",
        max_tokens=200
    )
    
    print(f"✅ Respuesta recibida:\n")
    print(f"{respuesta}\n")
except Exception as e:
    print(f"❌ Error: {str(e)}")

print("=" * 60)
print("\n🧪 PRUEBA 2: Chat con Historial\n")

try:
    messages = [
        {
            "role": "system",
            "content": "Eres un asistente matemático. Responde de forma concisa."
        },
        {
            "role": "user",
            "content": "¿Cuánto es 15 × 23?"
        }
    ]
    
    respuesta = ejecutar_chat_inteligente(messages, max_tokens=150)
    
    print(f"✅ Respuesta recibida:\n")
    print(f"{respuesta['content']}\n")
except Exception as e:
    print(f"❌ Error: {str(e)}")

print("=" * 60)
print("\n📊 ESTADÍSTICAS DEL SISTEMA:\n")

stats = obtener_estadisticas()

print("🔧 Configuración:")
print(f"   • Groq API Keys: {stats['configuracion']['groq_keys']}")
print(f"   • OpenRouter: {'✅ Configurado' if stats['configuracion']['openrouter_configured'] else '❌ No configurado'}")
print(f"   • Modelos disponibles: {stats['configuracion']['modelos_disponibles']}")

if stats['modelos']:
    print("\n📈 Uso de modelos:")
    for modelo, data in stats['modelos'].items():
        print(f"\n   {modelo}:")
        print(f"     • Usos: {data['usos']}")
        print(f"     • Éxitos: {data['exitos']}")
        print(f"     • Fallos: {data['fallos']}")
        print(f"     • Tiempo promedio: {round(data['tiempo_promedio'])}ms")
        print(f"     • Tokens usados: {data['tokens_usados']}")
else:
    print("\n⚠️ No hay estadísticas aún (primer uso)")

print("\n" + "=" * 60)
print("\n✨ Prueba completada! El sistema está funcionando correctamente.\n")
